bin/portal_utils/orbit_interface.py

```diff
@@ -0,0 +1,352 @@
+            orbit_group_member_query_url = \
+                "%s/getProjectMembers?groupname=%s" % \
+                (self._base_orbit_url, group_name)
+            orbit_member_info_raw = \
+                self.get_curl_output(orbit_group_member_query_url)
+            users = self.parse_group_members(orbit_member_info_raw)
+            for user in users:
+                orbit_users.add(user)
+                group_info['users'] = users
+        return orbit_groups, list(orbit_users)
+
+    # Parse results from getGroupsAndUsers call
+    def parse_group_user_info(self, orbit_info_raw):
+        groups = {}
+        orbit_info = xml.dom.minidom.parseString(orbit_info_raw)
+        group_nodes = orbit_info.getElementsByTagName('Group')
+        user_nodes = orbit_info.getElementsByTagName('User')
+        for group_node in group_nodes:
+            group_name = group_node.getAttribute('groupname')
+            group_admin = group_node.getAttribute('admin')
+            groups[group_name] = {'admin' : group_admin, 'users' : []}
+        for user_node in user_nodes:
+            user_name = user_node.getAttribute('username')
+            group_name = user_node.getAttribute('groupname')
+            if group_name not in groups:
+                print "Group %s for user %s not defined" % \
+                    (group_name, user_name)
+            group = groups[group_name]['users'].append(user_name)
+        return groups
+
+    # Parse results from getProjectMembers call
+    def parse_group_members(self, orbit_info_raw):
+        users = []
+        orbit_info = xml.dom.minidom.parseString(orbit_info_raw)
+        user_nodes = orbit_info.getElementsByTagName("User")
+        for user_node in user_nodes:
+            user_name = user_node.getAttribute('username')
+            users.append(user_name)
+        return users
+
+    # Save group/admin/user data in LDIF
+    def saveUser(self, user_ldif_data):
+    
+        save_user_url = "%s/saveUser" % self._base_orbit_url 
+
+        out_file, out_filename = tempfile.mkstemp()
+        os.write(out_file, user_ldif_data)
+        os.close(out_file)
+
+        ldif_arg = "ldif=@%s" % out_filename
+        output = ""
+        try:
+            output = subprocess.check_output(["curl", "-k", "-PUT", "-H", 
+                                              "Content-type: multipart/form-data", 
+                                              "-F", ldif_arg, save_user_url],
+                                             stderr=self._DEVNULL)
+        except subprocess.CaledProcessError as e:
+            print "Error invoking curl LDIF saveUser command: %s: Error %s" \
+                (save_user_url, e.returncode)
+            sys.exit(e.returncode)
+
+#        print "OUTPUT = %s" % output
+        error_code = self.find_error_code(output)
+        if error_code and error_code not in [CODES.ERROR1]:
+            print "Error in saveUser call: %s" % output
+            sys.exit(0)
```

Comment:
Should it really exit with code 0 when it gets an unknown error code? Doesn't 0 mean success? How about -1?